SML-Sandbox/ennate.egen.solutions.sml/src/main/java/ennate/egen/solutions/sml/utils/LexicalAnalyzerUtils.java

```diff
@@ -0,0 +1,105 @@
+package ennate.egen.solutions.sml.utils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.TreeMap;
+
+public class LexicalAnalyzerUtils {
+
+    public static double compareSentences(String sentence1, String sentence2) {
+        ArrayList<String> strings = new ArrayList<String>();
+        strings.add(sentence1);
+        strings.add(sentence2);
+        ArrayList<String[]> lowerCaseStringArrays = getSentenceArrayInLowerCase(strings);
+        HashSet<String> uniqueWordsInStrings = getUniqueWordsInStrings(lowerCaseStringArrays);
+        TreeMap<String, Integer> frequencyMap1 = getFrequencyOfWords(uniqueWordsInStrings, lowerCaseStringArrays.get(0));
```

Comment:
You can combine the frequency of words counting and the counting of unique words into one step. Essentially you do not have to traverse the sentences twice.